feature/runtime/testing/run.go

```diff
@@ -0,0 +1,31 @@
+package testing
+
+import (
+	"fmt"
+	"github.com/drognisep/systest/feature/runtime/errors"
+	lua "github.com/yuin/gopher-lua"
+)
+
+func run(tstate *State) func(l *lua.LState) int {
+	return func(l *lua.LState) int {
+		testName := l.ToString(1)
+		if len(testName) == 0 {
+			errors.LError(l, "empty test name")
+			return 0
+		}
+		fn := l.ToFunction(2)
+		p := lua.P{
+			Fn:      fn,
+			NRet:    0,
+			Protect: true,
```

Comment:
The `Protect: true` bit here will prevent errors from propagating beyond the `run` function. So any error that happens in the passed closure will be reported here and result in a failure in the test.